File: src/main/java/com/nightswatch/roomservice/entity/Room.java
package com.nightswatch.roomservice.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.GenericGenerator;

import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "rooms", indexes = {@Index(name = "idx_room_code", columnList = "roomCode", unique = true)})
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Room {

    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    @Column(name = "id", updatable = false, nullable = false)
    private UUID id;

    @Column(nullable = false, unique = true, length = 6)
    private String roomCode;

    @Column(nullable = false)
    private String hostId;

    private String currentVideoUrl;

    @Column(nullable = false)
    private boolean isActive = true;

    @Column(nullable = false, updatable = false)
    private Instant createdAt;

    @PrePersist
    public void prePersist() {
        if (createdAt == null) {
            createdAt = Instant.now();
        }
    }
}

File: nightswatch-client/src/types/room.ts
export interface Room {
  id: string
  roomCode: string
  hostId: string
  currentVideoUrl: string | null
  isActive: boolean
  streamPermissions: string[]
  createdAt: string
}

export interface Participant {
  userId: string
  online: boolean
}

export interface CreateRoomPayload {
  hostId: string
  currentVideoUrl?: string
}

export interface UpdateVideoPayload {
  currentVideoUrl: string
}
